components/shell/fish/patches/path_max.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/src/wutil.cpp b/src/wutil.cpp
+index 5b69c47f..9c4bedd2 100644
+--- a/src/wutil.cpp
++++ b/src/wutil.cpp
+@@ -139,7 +139,7 @@ bool wreaddir_for_dirs(DIR *dir, wcstring *out_name) {
+ const wcstring wgetcwd() {
+     wcstring retval;
+ 
+-    char *res = getcwd(NULL, 0);
++    char *res = getcwd(NULL, PATH_MAX);
```

Comment:
Don't know. https://github.com/fish-shell/fish-shell/issues/3340#issuecomment-269692610 says just "PATH_MAX" and it fixed the `getcwd() failed with errno 22/Invalid argument` error for me